Scripts/Python/psnlYeeshaPageChanges.py

```diff
@@ -269,18 +265,24 @@ def TimeToGrow(self):
 
     def UpdateState(self, state, size, SDLVar, AgeVault, sizechanged):
         #~ PtDebugPrint("No one else is here. Affecting any YP changes you've queued.")
-        if state == 3:
-            state = 2
-            PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 3 to 2." % ("YeeshaPage" + str(PageNumber.value)))
-            SDLVar.setInt( (size * 10) + state)
-            AgeVault.updateAgeSDL(self.ageSDL) 
-            
-        elif state == 4 or state > 4:
-            state = 1
-            PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 4 to 1." % ("YeeshaPage" + str(PageNumber.value)))
-            SDLVar.setInt( (size * 10) + state)
-            AgeVault.updateAgeSDL(self.ageSDL)
-        elif sizechanged:
-            SDLVar.setInt( (size * 10) + state)
-            AgeVault.updateAgeSDL(self.ageSDL) 
-        return state
+        if AgeVault := ptAgeVault():
+            if ageSDL := AgeVault.getAgeSDL():
+                if state == 3:
+                    state = 2
+                    PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 3 to 2." % ("YeeshaPage" + str(PageNumber.value)))
+                    SDLVar.setInt( (size * 10) + state)
+                    AgeVault.updateAgeSDL(ageSDL) 
+                    
+                elif state == 4 or state > 4:
+                    state = 1
+                    PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 4 to 1." % ("YeeshaPage" + str(PageNumber.value)))
+                    SDLVar.setInt( (size * 10) + state)
+                    AgeVault.updateAgeSDL(ageSDL)
+                elif sizechanged:
+                    SDLVar.setInt( (size * 10) + state)
+                    AgeVault.updateAgeSDL(ageSDL) 
+                return state
+            else:
+                PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the ageSDL. ageSDL = %s" % ( ageSDL))
+        else:
+            PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the Vault.")
```

Comment:
It looks like we already have the age vault, so we can simplify this by:

```suggestion
        if ageSDL := AgeVault.getAgeSDL():
            if state == 3:
                state = 2
                PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 3 to 2." % ("YeeshaPage" + str(PageNumber.value)))
                SDLVar.setInt( (size * 10) + state)
                AgeVault.updateAgeSDL(ageSDL) 
            elif state == 4 or state > 4:
                state = 1
                PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 4 to 1." % ("YeeshaPage" + str(PageNumber.value)))
                SDLVar.setInt( (size * 10) + state)
                AgeVault.updateAgeSDL(ageSDL)
            elif sizechanged:
                SDLVar.setInt( (size * 10) + state)
                AgeVault.updateAgeSDL(ageSDL) 
            return state
        else:
            PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the ageSDL. ageSDL = %s" % ( ageSDL))
    else:
        PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the Vault.")
```